--- primecoin-20131127/src/main.cpp~	2013-11-27 13:35:16.000000000 +0100
+++ primecoin-20131127/src/main.cpp	2013-12-04 14:58:01.574592313 +0100
@@ -4756,7 +4756,7 @@
             }
             else
             {
-#ifdef __GNUC__
+#if defined __GNUC__ && ! defined __i486__
                 // Use atomic increment
                 __sync_add_and_fetch(&nPrimeCounter, nPrimesHit);
                 __sync_add_and_fetch(&nTestCounter, nTests);
